arm_robots/src/arm_robots/robot.py

```diff
@@ -90,7 +91,9 @@ def set_block(self, block: bool):
         self.block = block
 
     def get_move_group_commander(self, group_name: str) -> moveit_commander.MoveGroupCommander:
-        move_group = moveit_commander.MoveGroupCommander(group_name, ns=self.robot_namespace)
+        if group_name not in self._move_groups:
+            self._move_groups[group_name] = moveit_commander.MoveGroupCommander(group_name, ns=self.robot_namespace)
+        move_group = self._move_groups[group_name]
```

Comment:
I still want to add the behavior where all groups are allocated on connect. I have a lot of issues in simulation where things don't connect when I'm messing with simulation pausing/unpausing/time, etc...

So the simplest thing would be combine what you have here with the bit from my PR where it loops over all groups and calls `get_move_group_commander`. Would that work?

Reply:
Let me know what you think about this compromise: Add an option to preload all the of the move groups commanders

Reply:
that's great. I would rename it to `preload_move_groups` though